src/overlay/overlay.scss

```diff
@@ -0,0 +1,379 @@
+}
+
+.Overlay {
+  display: flex;
+  min-width: 192px;
+  flex-direction: column;
+  background-color: var(--color-canvas-default);
+  border-radius: $primer-borderRadius-large;
+  box-shadow: var(--color-overlay-shadow);
+  opacity: 1;
+
+  &.Overlay--height-auto {
+    height: auto;
+  }
+
+  &.Overlay--height-xsmall {
+    height: min(192px, 100% - 2rem);
+  }
+
+  &.Overlay--height-small {
+    height: min(256px, 100% - 2rem);
+  }
+
+  &.Overlay--height-medium {
+    height: min(320px, 100% - 2rem);
+  }
+
+  &.Overlay--height-large {
+    height: min(432px, 100% - 2rem);
+  }
+
+  &.Overlay--height-xlarge {
+    height: min(600px, 100% - 2rem);
+  }
+
+  &.Overlay--width-auto {
+    width: auto;
+  }
+
+  &.Overlay--width-small {
+    width: min(256px, 100% - 2rem);
+  }
+
+  &.Overlay--width-medium {
+    width: min(320px, 100% - 2rem);
+  }
+
+  &.Overlay--width-large {
+    width: min(480px, 100% - 2rem);
+  }
+
+  &.Overlay--width-xlarge {
+    width: min(640px, 100% - 2rem);
+  }
+
+  &.Overlay--width-xxlarge {
+    width: min(960px, 100% - 2rem);
+  }
+
+  &.Overlay--motion-scaleFade {
+    @media screen and (prefers-reduced-motion: no-preference) {
+      animation: 200ms cubic-bezier(0.33, 1, 0.68, 1) 0s 1 normal none running motion--scaleFade;
+    }
+  }
+
+  @keyframes motion--scaleFade {
+    0% {
+      opacity: 0;
+      transform: scale(0.5);
+    }
+
+    100% {
+      opacity: 1;
+      transform: scale(1);
+    }
+  }
+}
+
+.Overlay-header {
+  z-index: 1;
+  display: flex;
+  flex-direction: column;
+
+  &.Overlay-header--divided {
+    padding-bottom: $spacer-2;
+    // stylelint-disable-next-line primer/box-shadow
+    box-shadow: inset 0 #{-$border-width} var(--color-border-default);
+
+    + .Overlay-body {
+      padding-top: $spacer-3;
+    }
+  }
+
+  &.Overlay-header--large {
+    .Overlay-headerContentWrap {
+      .Overlay-titleWrap {
+        gap: $spacer-2;
+
+        .Overlay-title {
+          font-size: $h3-size;
+        }
+
+        .Overlay-description {
+          font-size: $body-font-size;
+        }
+      }
+    }
+  }
+
+  .Overlay-headerContentWrap {
+    display: flex;
+    align-items: flex-start;
+    gap: $spacer-2;
+    padding: $spacer-2 $spacer-2 0 $spacer-2;
+
+    .Overlay-actionWrap {
+      display: flex;
+      flex-direction: row;
+      gap: $spacer-2;
+    }
+
+    .Overlay-titleWrap {
+      display: flex;
+      padding: ($spacer-2 * 0.75) 0 ($spacer-2 * 0.75) $spacer-2;
+      flex-direction: column;
+      flex-grow: 1;
+      gap: $spacer-1;
+
+      .Overlay-title {
+        margin: 0;
+        font-size: $body-font-size;
+        font-weight: $font-weight-bold;
+      }
+
+      .Overlay-description {
+        margin: 0;
+        font-size: $font-size-small;
+        font-weight: $font-weight-normal;
+        color: var(--color-fg-muted);
+      }
+    }
+  }
+}
+
+// generic body content slot
+.Overlay-body {
+  padding: $spacer-3;
+  padding-top: 0;
+  overflow-y: scroll;
```

Comment:
Should we change this to `auto` so that it doesn't add an empty "scrollbar container" when there is nothing to scroll?

```suggestion
  overflow-y: auto;
```

![Screen Shot 2022-04-27 at 12 12 38](https://user-images.githubusercontent.com/378023/165432561-56ac321f-d3a1-45e9-bd76-5f3baf916ba4.png)